packages/social-auth/src/Auth.ts

```diff
@@ -17,14 +17,19 @@ const WEB3_AUTH_CLIENT_ID =
 export abstract class Auth {
   abstract idpName: IDP;
   abstract clientId: string;
+  redirectSurface: RedirectSurface;
+
+  constructor(redirectSurface: RedirectSurface) {
+    this.redirectSurface = redirectSurface;
+  }
 
   /* istanbul ignore next */
   protected async getTorusClient(): Promise<CustomAuth> {
     const torus = new CustomAuth({
       web3AuthClientId: WEB3_AUTH_CLIENT_ID,
       baseUrl: "https://umamiwallet.com/auth/v2.2.0/",
       redirectPathName: "redirect.html",
-      redirectToOpener: true,
+      redirectToOpener: this.redirectSurface as any as boolean,
```

Comment:
a comment on why we do that would be helpful

Reply:
Good point. Added now